lib/Monad_Commute.thy

```diff
@@ -0,0 +1,195 @@
+(*
+ * Copyright 2023, Proofcraft Pty Ltd
+ * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ *)
+
+(* The monad_commute predicate + lemmas. *)
+
+theory Monad_Commute
+  imports
+    Monad_Equations
+    Monad_Lists (* for mapM_x *)
+begin
+
+
+definition monad_commute where
+  "monad_commute P a b \<equiv>
+   \<forall>s. (P s \<longrightarrow> ((do x\<leftarrow>a;y\<leftarrow>b; return (x, y) od) s) = ((do y\<leftarrow>b;x\<leftarrow>a; return (x, y) od) s))"
+
+
```

Comment:
extra newline?

Reply:
This one actually makes sense, separating the definition from the rest.